app/admin/dashboard.rb

```diff
@@ -47,6 +47,104 @@ def set_dashboard_values
+      @new_users_by_month = []
+      @month_and_year_array = []
+      start_date = Date.today.prev_year.beginning_of_month
+      end_date = Date.today
+
+      while start_date <= end_date
+        beg_of_month = start_date.beginning_of_day
+        end_of_month = start_date.end_of_month.end_of_day
+        month_and_year = Date::MONTHNAMES[start_date.month] + " #{ start_date.year.to_s}"
+        # Get practice views by month
+        @approved_enabled_published_practices.each do |p|
+          practice_visits_by_month = []
+          pr_visit_ct = Ahoy::Event.where_props(practice_id: p[:id]).where(time: beg_of_month...end_of_month).count
+          practice_visits_by_month << pr_visit_ct
+          practice_visits_by_month.each do |visit_count|
+            @practice_views_array << [p.id, visit_count]
+          end
+        end
+
+        # Get site visits by month
+        site_visit_by_ip = Ahoy::Event.where(name: 'Site visit').where("properties->>'ip_address' is not null").where(time: beg_of_month..end_of_month).group("properties->>'ip_address'").count
+        site_visit_ct = site_visit_by_ip.sum {|_k, v| v}
+        @site_visits_by_month << [month_and_year, site_visit_ct]
+
+        # Get new users by month
+        new_user_count = User.where(created_at: beg_of_month..end_of_month).count
+        @new_users_by_month << new_user_count
+
+        # Get total users by month
+        total_user_count = User.where(current_sign_in_at: beg_of_month..end_of_month).count
+        @total_users_by_month << total_user_count
+
+        @month_and_year_array << [month_and_year]
+        start_date += 1.months
+      end
+
+      # Assign the monthly practice views to the correct practice based on its id
+      @practice_views_by_month = []
+      @approved_enabled_published_practices.each do |p|
+        @practice_views_array.each do |array|
+          if p.id === array.first
+            @practice_views_by_month << [p.name, array.last]
+          end
+        end
+      end
+
+
+      # Add user stats hash to make it easier to format spreadsheet data
+      @user_statistics = {
+          new_users: @new_users_by_month,
+          total_users: @total_users_by_month
+      }
+    end
+
+    def add_header_row_for_month_and_year(sheet, first_column_text, array, row_style)
+      sheet.add_row [
+          "#{first_column_text}",
+          array[0].join(' '),
+          array[1].join(' '),
+          array[2].join(' '),
+          array[3].join(' '),
+          array[4].join(' '),
+          array[5].join(' '),
+          array[6].join(' '),
+          array[7].join(' '),
+          array[8].join(' '),
+          array[9].join(' '),
+          array[10].join(' '),
+          array[11].join(' '),
+          array[12].join(' ')
+      ], style: row_style
+    end
+
+    def add_monthly_array_values_to_columns(hash, sheet, row_style)
+      hash.each do |key, value|
+        sheet.add_row [
+            key.to_s.tr!('_', ' ').titleize,
```

Comment:
i think we can clean this up similarly to the above comment

Reply:
Sure can! Code has been updated! Thank you!